1.0-Upgrade.md

```diff
@@ -0,0 +1,108 @@
+# kubernetes-deploy 0.30.0 -> krane 1.0.0 migration guide
+
+**THIS MIGRATION GUIDE IS A WORK IN PROGRESS. THINGS CAN STILL CHANGE BEFORE 1.0.0**
+
+`kubernetes-deploy` was renamed `krane` when version 1.0.0 was released. Version 1.0.0 introduced new features and [breaking changes](CHANGELOG.md). This guide will help you transition to version 1.0.0 as smoothly as possible.
+
+**TL;DR**:
+* The command-line interface was redesigned; besides the name change, there are breaking changes in several flags.
+* There are breaking changes in the public API (such as the renaming of the `KubernetesDeploy` namespace to `Krane`, and the change in default values for different arguments of the public interface).
+* StatsD metrics will now be generated with the `krane` prefix.
+* `kubernetes-deploy` (now `krane deploy`) / `DeployTask` can no longer deploy global (non-namespaced) resources. A new command called `krane global-deploy` and a related class called `GlobalDeployTask` were added to replace that feature.
+* If you attempt to install two gems that have conflicting executables, `gem install` will warn you but the most recently installed one will win. This means that you can run both `kubernetes-deploy` 0.30.0 and `krane` 1.0.0 side by side by doing:
+
+```bash
+gem install kubernetes-deploy
+gem install krane
+```
+
+## Public API changes
+
+The only breaking change in the public API is the renaming of the `KubernetesDeploy` namespace to `Krane`. Besides that, the API of the major public classes (`DeployTask`, `RenderTask`, `RunnerTask`, and `RestartTask`) didn't change at all between 0.30.0 and 1.0.0. If you're curious about this API, check the comment-based docs on these classes or the [rendered documentation at RubyGems.org](https://www.rubydoc.info/gems/kubernetes-deploy/1.0.0/KubernetesDeploy/DeployTask).
```

Comment:
I don't think this is true, args and defaults changed. We did also change functionality by  removing the ability to deploy global resources)

Reply:
Did args really change between `kubernetes-deploy` 0.30.0 and `krane 1.0.0`? I mean, their names will change but the default values are already different now aren't they?

Reply:
I think the goal of the guide it help someone move from  `KubernetesDeploy::XTask` - > `Krane::XTask` as well as `exe/kubernetes-x` -> `exe/krane x` . An example of a default value change would be erb rendering.

Reply:
That's an example of a _flag_ default value change; the default value in the code is still `false`:

https://github.com/Shopify/kubernetes-deploy/blob/6c6f930cecf3fda21b6fb57cca4f7ab39455d0c9/lib/krane/deprecated_deploy_task.rb#L128-L130

My point is that people moving from `KubernetesDeploy::DeployTask` to `Krane::DeployTask` will not actually have to change much **if they are on a recent version**. For instance, global deploys were disabled by default in 0.30.0.

Reply:
So it's true right now because we've intentionally forced ourselves into backwards compatibility. But, once we cut 1.0.0-pre.1 it wont be true. Are you thinking the release of 1.0.0-pre.1 will change the wording? Or the first PR to actually make a change will be responsible for updating this?

Reply:
> Or the first PR to actually make a change will be responsible for updating this?

That's what I'm thinking. I plan to have this merged and then rebase https://github.com/Shopify/kubernetes-deploy/pull/619 and have that PR document particular changes. It might be easier to keep track of all the changes we're doing that way. Does that make sense?

Reply:
How about: `The only breaking change in the public API, so far, is` . I don't want someone to read this today and see the 1.0 and thing ok cool. Not realizing this is a living doc.

Reply:
Makes sense. Updated.